lib/tests/streamlit/net_util_test.py

```diff
@@ -37,6 +37,22 @@ def test_get_external_ip(self):
             m.get(net_util._AWS_CHECK_IP, exc=requests.exceptions.ConnectTimeout)
             self.assertEqual(None, net_util.get_external_ip())
 
+    def test_get_external_ip_use_http_by_default(self):
+        # Test that by default we do call http endpoint
+        with requests_mock.mock() as m:
+            m.get(net_util._AWS_CHECK_IP, text="1.2.3.4")
+            m.get(net_util._AWS_CHECK_IP_HTTPS, text="5.6.7.8")
+            self.assertEqual("1.2.3.4", net_util.get_external_ip())
+            # Test that we do only one http call
+            self.assertEqual(m.call_count, 1)
+
+    def test_get_external_ip_https(self):
+        # Test that if http fails, we try https
```

Comment:
nit: same thing here -- we can probably drop these comments since the test names are descriptive enough + maybe edit the second test name a bit